src/utils/formValidation.ts

```diff
@@ -0,0 +1,93 @@
+import { ValidatorRules, ValidatorResult, FormFieldName } from "../types/form";
+
+/* 공통 유효성 체크 */
+export const validators: ValidatorRules = {
+  image: (value) => {
+    if (value)
+      return {
+        isValid: false,
+        message: "*이미지 등록은 최대 1개까지 가능합니다.",
+      };
+    return { isValid: true, message: "" };
+  },
+  title: (value) => {
+    if (!value) return { isValid: false, message: "상품명을 입력해주세요" };
+    return { isValid: true, message: "" };
+  },
+  content: (value) => {
+    if (!value) return { isValid: false, message: "상품 소개를 입력해주세요" };
+    return { isValid: true, message: "" };
+  },
+  price: (value) => {
+    const unformattedVal = value.replace(/,/g, "");
+    const regex = /^[1-9]\d*$/;
+    if (!value) return { isValid: false, message: "판매 가격을 입력해주세요" };
+    if (!regex.test(unformattedVal))
+      return {
+        isValid: false,
+        message: "판매 가격 형식을 올바르게 입력해주세요",
+      };
+
+    return { isValid: true, message: "" };
+  },
+  tagList: (value) => {
+    if (value.length === 0)
+      return { isValid: false, message: "태그를 입력해주세요" };
+    return { isValid: true, message: "" };
+  },
+  inquiry: (value) => {
+    if (!value) return { isValid: false, message: "문의 내용을 입력해주세요" };
+    return { isValid: true, message: "" };
+  },
+  email: (value) => {
+    const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/i;
+    if (!value) return { isValid: false, message: "이메일을 입력해주세요" };
+    if (!regex.test(value))
+      return {
+        isValid: false,
+        message: "잘못된 이메일 형식입니다",
+      };
+
+    return { isValid: true, message: "" };
+  },
+  password: (value) => {
+    if (!value) return { isValid: false, message: "비밀번호를 입력해주세요" };
+    if (value.length < 8)
+      return {
+        isValid: false,
+        message: "비밀번호를 8자 이상 입력해주세요",
+      };
+
+    return { isValid: true, message: "" };
+  },
+  passwordCheck: (value, password) => {
+    if (!password)
+      return { isValid: false, message: "비밀번호를 먼저 입력해주세요" };
+    if (value.length < 8)
+      return { isValid: false, message: "비밀번호를 8자 이상 입력해주세요" };
+    if (value !== password)
+      return { isValid: false, message: "비밀번호가 일치하지 않습니다" };
+    return { isValid: true, message: "" };
+  },
+  nickname: (value) => {
+    if (!value) return { isValid: false, message: "닉네임을 입력해주세요" };
+    return { isValid: true, message: "" };
+  },
+};
+
+export const validateInput = <T extends FormFieldName>(
+  validatorType: T,
+  ...values: Parameters<ValidatorRules[T]>
+): ValidatorResult => {
+  const errorValidator = () => ({
+    isValid: false,
+    message: "유효성 체크가 정의되지 않았습니다.",
+  });
+
+  const validator =
+    (validators[validatorType] as (
+      ...args: Parameters<ValidatorRules[T]>
+    ) => ValidatorResult) || errorValidator;
+  const { isValid, message } = validator(...values);
+  return { isValid, message };
+};
```

Comment:
폼 유효성 체크 (formValidation)파일에서 validateInput에서 여러 개의 값을 받기 위해 args를 사용하였는데, 타입을 지금처럼 작성해도 될까요?

Reply:
잘 하셨는데, 타입을 조금 복잡하게 쓰시는 편인것 같아서, ValidatorParams 타입을 추가해 validateInput에서 제네릭으로 각 필드별 타입을 안전하게 추론하게끔 개선하면 좋겠네요! 👍 

```ts
// 각 필드별 유효성 검사 함수의 매개변수 타입 정의
export type ValidatorParams = {
  image: [value: string];
  title: [value: string];
  content: [value: string];
  price: [value: string];
  tagList: [value: string[]];
  inquiry: [value: string];
  email: [value: string];
  password: [value: string];
  passwordCheck: [value: string, password: string];
  nickname: [value: string];
};
```

```suggestion
export const validateInput = <T extends FormFieldName>(
  validatorType: T,
  ...values: ValidatorParams[T]
): ValidatorResult => {
  const validator = validators[validatorType] as (
    ...args: ValidatorParams[T]
  ) => ValidatorResult;
  return validator(...values);
};
};
```

Reply:
넵! 이 부분 타입 추가하여 수정하겠습니다